pkg/apis/operator/v1/types.go

```diff
@@ -100,6 +100,27 @@ type InstallationSpec struct {
 	// field.
 	// +optional
 	NodeUpdateStrategy appsv1.DaemonSetUpdateStrategy `json:"nodeUpdateStrategy,omitempty"`
+
+	// ComponentResources can be used to customize the resource requirements for each component.
+	// +optional
+	ComponentResources []*ComponentResource `json:"componentResources,omitempty"`
+}
+
+// ComponentName CRD enum
+type ComponentName string
+
+const (
+	ComponentNameNode            ComponentName = "Node"
+	ComponentNameTypha           ComponentName = "Typha"
+	ComponentNameKubeControllers ComponentName = "KubeControllers"
+)
+
+// The ComponentResource struct associates a ResourceRequirements with a component by name
+// +k8s:openapi-gen=true
+type ComponentResource struct {
+	// +kubebuilder:validation:Enum=Node;Typha;KubeControllers
```

Comment:
We should have a description on both of these fields.

Reply:
agree, done